examples/plot_incr_kmeans.py

```diff
@@ -0,0 +1,274 @@
+samples are less likely to be selected nearby an already selected sample.
+
+This notebook introduces the principle behind Incremental KMeans and runs
+a small benchmark on generated data.
+
+"""
+
+from sklearn.cluster import MiniBatchKMeans
+from sklearn.metrics import pairwise_distances_argmin_min
+from cardinal.kmeans import IncrementalMiniBatchKMeans
+from sklearn.datasets import make_blobs
+from matplotlib import pyplot as plt
+from scipy.stats import ttest_rel
+import numpy as np
+
+##############################################################################
+# Let us first define the parameters of our experiment. We will generate a
+# dataset from 8 clusters and we will try to keep 4 clusters fixed. We do
+# this example in 2 dimensions for visualization purposes.
+
+n_clusters = 8
+n_fixed_clusters = 4
+
+X, y, centers = make_blobs(centers=n_clusters, return_centers=True,
+                           n_features=2, random_state=2)
+
+##############################################################################
+# We define plotting functions. This function allows to plot the data colored
+# by clusters, and allows to displays fixed cluster centers in red and regular
+# centers in blue. We use it to display the ground truth.
+
+def plot_clustering(label, y_pred, centers, fixed_centers=[], inertia=None):
+    colors = ['C{}'.format(i) for i in y_pred]
+    plt.scatter(X[:, 0], X[:, 1], c=colors)
+    plt.scatter(centers[:, 0], centers[:, 1], c='blue', edgecolors='k',
+                marker='P', s=200)
+    if len(fixed_centers) > 0:
+        plt.scatter(fixed_centers[:, 0], fixed_centers[:, 1],
+                    c='red', edgecolors='k', marker='P', s=200)
+    plt.xticks([])
+    plt.yticks([])
+    plt.tight_layout(rect=[0, 0.03, 1, 0.95])
+    if inertia:
+        label = label + ", intertia={}".format(inertia)
+    plt.title(label)
+    plt.show()
+
+plot_clustering('Ground truth', y, centers)
+
+##############################################################################
+# We run a regular MiniBatchKMeans. KMeans would be more fitted for this kind
+# of small dataset but we are aiming at using Increment KMeans on large
+# datasets so its implementation relies on MiniBatchKMeans.
+
+kmeans = MiniBatchKMeans(n_clusters=8, random_state=2)
+kmeans.fit(X)
+plot_clustering('KMeans', kmeans.predict(X), kmeans.cluster_centers_,
+                inertia=kmeans.inertia_)
+
+
+##############################################################################
+# We now consider that we are aware of 4 of the 8 clusters. We fix them in
+# the IncrementalMiniBatchKMeans so that they are strictly enforced.
+
+ikmeans = IncrementalMiniBatchKMeans(n_clusters=8, random_state=2)
+ikmeans.fit(X, fixed_cluster_centers=centers[:n_fixed_clusters])
+plot_clustering(
+    'Incremental KMeans', ikmeans.predict(X),
+    kmeans.cluster_centers_[n_fixed_clusters:],
+    fixed_centers=kmeans.cluster_centers_[:n_fixed_clusters],
+    inertia=kmeans.inertia_)
```

Comment:
As discussed, I think this is meant to be 
```suggestion
plot_clustering(
    'Incremental KMeans', ikmeans.predict(X),
    centers[n_fixed_clusters:],
    fixed_centers=centers[:n_fixed_clusters],
    inertia=ikmeans.inertia_)
```